### PR TITLE
Allow to store custom map themes (authenticathed users)

### LINKED ISSUE #539: Save new map theme from  G3W-SUITE APP of current Catalog TOC state
### Checklist

- [X] I've searched through the [current issues](https://github.com/g3w-suite/g3w-client/issues) to make sure this feature hasn't been requested already.

### Motivation

It could be useful to save a current state of TOC as a new map theme as in QGIS

Screenshot from 2024-01-09 12-00-06

### Suggested solution

Add save button on map on CHOOSE THEME SECTION
Screenshot from 2024-01-09 12-01-51

### Alternatives considered

_No response_

### LINKED ISSUE #614: Check user input (field constraints) while editing the attribute and not after saving the form
### Checklist

- [x] I've searched through the [current issues](../issues) to make sure this feature hasn't been requested already.

### Motivation

It would be very useful if, during the attribute editing phase, a check is active on the fields relating to the content entered by the user which informs me (as for the other constraints on the fields) for example if a user enters text in a numeric field

It isn't currently there and I only notice the error when saving

### Suggested solution

Walter probably also has to forward the field type and Francesco check the consistency of the inserted content

### Alternatives considered

_No response_

### LINKED ISSUE #615: Choosing of Atlas item to print throws: "TypeError: can't access property 'top'"
### Checklist

- [X] I've searched through the existing [issues](https://github.com/g3w-suite/g3w-client/issues) and this bug has never been reported before

### Subject of the issue

An error occurs in the console while choosing the item to print for an Atlas:

Screenshot_20240424_104520-1

### Steps to reproduce

1. Create a QGIS project with a Atlas layout without map items.
2. Updload on G3W-SUITE 
3. Try to get a print from G3W-CLIENT choosing the layout without map items.

### Environment

[g3wsdk.info]

- g3w-admin: __3.7.x-20240422122352__
- g3w-client: __3.10.0-alpha.1__
    - editing: __3.8.0-alpha.1__
    - qtimeseries: __undefined__
- browser: __Firefox 126.0__
- operating system: __Linux 64-bit__

### Link to your project

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by volterra79**
## Depends on: https://github.com/g3w-suite/g3w-admin/pull/813 

Closes: #539, fixes: #614, fixes: #615

Screenshot from 2024-04-17 12-26-48

### CODE REVIEW

**Review by Raruto — APPROVED** [review]
The JS code is quite good, in the related admin PR I reported some tests that could be further improved:

- https://github.com/g3w-suite/g3w-admin/pull/813#issuecomment-2063315590
- https://github.com/g3w-suite/g3w-admin/pull/813#issuecomment-2063829483

File: src/app/core/project/project.js
Comment (Raruto) [inline src/app/core/project/project.js:1]: @volterra79 please double check this merge commit too, there were some conflicts

**Review by Raruto — CHANGES_REQUESTED** [review]

**Comment by leolami** [comment]
If any theme is saved in the project, I can not save any new user themes

For esample in this project

https://dev.g3wsuite.it/it/map/enel/qdjango/40/

**Comment by leolami** [comment]
1. I create and save a theme
2. A switch on a new layer
3. I upload the previous theme
4. The messagge is 'Theme upload with success"

But the new layer is NOT added to the theme

**Review by Raruto — APPROVED** [review]

File: src/utils/validators/index.js [hunk 1]
```diff
@@ -53,6 +71,26 @@ const InputValidators = {
...
+    /**
+     * @since 3.10.0
+     * @param options
+     */
+    chart(options) {
+      this.options = options;
+      this.validate = function(value) {
+        return value && 1 === `${value}`.length;
+      }
+    },
```
Comment (Raruto) [inline src/utils/validators/index.js]: @volterra79 is this an oversight? ie. `chart(options)` → `char(options)`

Comment (volterra79) [reply src/utils/validators/index.js]: @Raruto rigth. It's a misspelling

**Review by Raruto — APPROVED** [review]

**Pull request merged**